packages/geoprocessing/src/toolbox/geoblaze/rasterStats.ts

```diff
@@ -83,31 +83,44 @@ export const rasterStats = async (
 
   try {
     // If categorical raster, use histogram to calculate valid cells
-    statsByBand = categorical
-      ? (
-          await geoblaze.histogram(raster, projectedFeat, {
-            scaleType: "nominal",
-          })
-        ).map((h: any) => {
+    if (categorical) {
+      const histogram = await geoblaze.histogram(raster, projectedFeat, {
+        scaleType: "nominal",
+      });
+
+      // If no overlap
+      if (!histogram.length) {
+        if (!categoryMetricValues || categoryMetricValues.length === 0) {
+          statsByBand = [{ histogram: {} }];
+        } else {
+          let hist = {};
+          categoryMetricValues.forEach((c) => (hist[c] = 0));
+          statsByBand = [{ histogram: hist }];
+        }
```

Comment:
I think this default value setting should be taking place up in the defaultValues code top of file.